src/org/rust/lang/structure/RustFileTreeElement.kt

```diff
@@ -18,6 +16,14 @@ class RustStructureViewTreeElement(file: RustFileImpl) : PsiTreeElementBase<Rust
                 o.acceptChildren(this)
             }
 
+            override fun visitEnumItem(o: RustEnumItem) {
+                result += (RustStructureViewElement(o))
```

Comment:
right now it is used so that methods inside of `impl` blocks are added to the tree too, but once they are properly structured we don't need the recursiveness anymore